proxy/handler/request_handler.go

```diff
@@ -165,6 +165,40 @@ func (h *RequestHandler) HandleWebSocketRequest(iter route.EndpointIterator) {
 	h.reporter.CaptureWebSocketUpdate()
 }
 
+func (h *RequestHandler) SanitizeRequestConnection() {
+	connections := h.request.Header.Values("Connection")
+	for index, connection := range connections {
+		if connection != "" {
+			values := strings.Split(connection, ",")
+			connectionHeader := []string{}
+			bannedList := []string{
```

Comment:
hi @winkingturtle-vmw thanks for your PR!
1. We want to keep these headers and protect them from getting dropped as a hop-by-hop, right? I think it may be a bit confusing to call the list `bannedList` then. Maybe `keepHeaders` or `headerToKeep`?

2. I think it makes sense to not have this list hard-coded as it may be different depending on the organization where CF is deployed.
I suggest to make it configurable in the [config](https://github.com/cloudfoundry/gorouter/blob/main/config/config.go) and provide a [spec](https://github.com/cloudfoundry/routing-release/blob/develop/jobs/gorouter/spec) for it.

You may provide the hard-coded list here as a default value for the spec and also put it into the [default config](https://github.com/cloudfoundry/gorouter/blob/main/config/config.go#L318) so that it is active by default.

Reply:
Thanks @domdom82 for you feedback. I think the spec property is an option, but can I please ask you with the task coming up with a good name? 

I am happy to change the name from `bannedList` to anything folks think is more easily understood, but essentially those header values would be banned from Client side on `Connection` header and are subject to removal. I know for me personally, `headersToKeep` would make it sound like a property that would need to have every single header used in gorouter.

Reply:
@winkingturtle-vmw what about `connectionHeadersToKeep`? I think it fits well with `connectionHeader` on line 173 and expresses that these are headers on the connection we want to keep instead of drop.

Reply:
I think between the two options, I would prefer `headersToKeep` probably because the values in this field are to be removed from `Connection` header instead of kept. Alternatively, I could just call this variable in this function a `list` to avoid any confusion. That said, in order to make this a property of gorouter, we would still need to give it a name that I am not sure what to call.

Reply:
ok, then I'll make the call to name it `headersToKeep` for now. In the spec you can add more description that explains in detail what the thing does.

Reply:
The variable was no longer needed after adding the spec property. It's now called `HopByHopHeadersToFilter` property. Is that okay if we now resolve this conversation?